pkg/bloombuild/planner/planner.go

```diff
@@ -272,6 +278,12 @@ func (p *Planner) loadWork(
 				})
 			}
 
+			// Reset progress tracking metrics for this tenant
+			// NOTE(salvacorts): We will reset them multiple times for the same tenant, for each table, but it's not a big deal.
+			//                   Alternatively, we can use a Counter instead of a Gauge, but I think a Gauge is easier to reason about.
+			p.metrics.tenantTasksPlanned.WithLabelValues(tenant).Set(0)
+			p.metrics.tenantTasksCompleted.WithLabelValues(tenant).Set(0)
```

Comment:
nit: should we remove the metric with the tenant label completely, instead of just resetting it to 0?

Reply:
Also, we could then use a Counter

Reply:
> should we remove the metric with the tenant label completely, instead of just resetting it to 0?

AFAICT a metric cannot be explicitly removed. Can it?

> we could then use a Counter

That was my first idea but a counter would make it a bit difficult to see for a given iteration, how many "new" tasks would be created. It would require using `increase` but setting the interval is difficult. Instead, by using a gauge, you can see the current set of tasks.

Also note that the gauge doesn't change often, but only when the build is planned.